07_leukemia_patient/simulation_patient_cell_lines/workflow.sh

```diff
@@ -0,0 +1,389 @@
+  { \
+    current = \$1; \
+    current_ub_cb = \$28 \";\" \$27; \
+    current_ub = \$28; \
+    cmd = \"grep -q \" current \" r2.fastq\"; \
+    exit_code = system(cmd); \
+    if (exit_code == 0) { next; } \
+    if (current_ub != seen_ub && current_ub_cb != seen_ub_cb && current != seen) { \
+      print \"@\" current \";\" \$27 \";\" \$28 \"\\n\" \$10 \"\\n+\\n\" \$11; \
+    } \
+    seen = current; \
+    seen_ub_cb = current_ub_cb; \
+    seen_ub = current_ub; \
+  }"' >> r2.fastq
+
+echo "ABL extracted"
+
+# for unmapped reads: they don't have the gx tag, so the CB and UB will be at a different position (UB (23), CB (22) and 1 (read id)
+
+samtools view -b -f 4 "$STARSOLO_BAM" -@ "$NTHREADS" \
+| samtools view -@ "$NTHREADS" \
+| grep -v "CB:Z:-" \
+| grep -v "UB:Z:-" \
+| sort -k23,23 -k22,22 -k1,1 \
+| parallel --pipe --block 10M -j "$NTHREADS" \
+'awk -v seen="" -v current="" -v current_ub_cb="" -v seen_ub_cb="" -v current_ub="" -v seen_ub="" \
+  "BEGIN { OFS=\"\t\" } \
+  { \
+   current = \$1; \
+   current_ub_cb = \$23 \";\" \$22; \
+   current_ub = \$23; \
+   cmd = \"grep -q \" current \" r2.fastq\"; \
+   exit_code = system(cmd); \
+   if (exit_code == 0) { next; } \
+   if (current_ub != seen_ub && current_ub_cb != seen_ub_cb && current != seen) { \
+     print \"@\" current \";\" \$22 \";\" \$23 \"\\n\" \$10 \"\\n+\\n\" \$11; \
+   } \
+   seen = current; \
+   seen_ub_cb = current_ub_cb; \
+   seen_ub = current_ub; \
+  }"' >> r2.fastq
+
+echo "unmapped extracted"
+
+rm header.txt
+
+## Step 2: additonal CB / UB deduplication
+
+# Additional CB and UB deduplication if reads appended in different steps have same CB and UB
+
+awk 'NR%4==1{print}' r2.fastq | awk '{split ($0, a, /[;]/); print a[2]";"a[3]}' | sort | uniq -d > duplicated.txt
+
+echo 'remaining reads to deduplicate'
+
+wc -l duplicated.txt
+
+# obtain the entry in the .fastq file for the duplicated CB and UB. Only taking the second line as the unmapped will always be at that position
+
+grep -Ff duplicated.txt r2.fastq | awk 'NR % 2 == 1' > unmapped_reads.txt
+
+# remove the lines from that entry and generate new file --> only works if the wc -l of the unmapped_reads.txt is larger than 0, so need to add that condition
+
+awk 'NF {exit 1}' unmapped_reads.txt && mv r2.fastq deduplicated_r2.fastq || mv r2.fastq deduplicated_r2.fastq; while read -r id; do sed -i "/$id/,+3d" deduplicated_r2.fastq; done < unmapped_reads.txt
+
+rm duplicated.txt unmapped_reads.txt
+
+mv deduplicated_r2.fastq sorted_duplicate_r2.fastq
+
+pigz sorted_duplicate_r2.fastq
+
+echo "R2 file generated"
+
+## Step 3: running bwa aln
+
+mkdir -p $WD/bwa_aln/genome
+cd $WD/bwa_aln/genome
+
+# transcript reference generation for bwa aln
+
+wget http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/"$TRANSCRIPTOME".gz
+
+pigz --decompress *gz
+
+# ENSG00000097007: ABL1
+# ENSG00000186716: BCR
+
+# extracting all transcripts for ABL1 and BCR from the .fa transcriptome
+
```

Comment:
These are no gtf files, but rather transcript ids / transcript names, perhaps renaming the files?